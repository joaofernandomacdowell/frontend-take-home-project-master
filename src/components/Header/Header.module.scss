@import '../../styles/mixins';
@import '../../styles/variables.scss';

.header {
  background-color: $gray-600;
}

.title {
  font-size: 24px;
}
