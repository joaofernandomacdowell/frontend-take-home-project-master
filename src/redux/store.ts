/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import { throttle } from 'lodash';

import { loadState, saveState } from '../localStorage/localStorage';

import rootReducer from './reducers';

// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
const persistedState = loadState();

const store = createStore(
  rootReducer,
  persistedState,
  composeWithDevTools(applyMiddleware(thunk))
);

store.subscribe(
  throttle(() => {
    saveState({
      countriesState: store.getState().countriesState,
    });
  }, 1000)
);

export type AppState = ReturnType<typeof rootReducer>;

export default store;
