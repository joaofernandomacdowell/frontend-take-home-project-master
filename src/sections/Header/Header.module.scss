@import '../../styles/mixins';
@import '../../styles/variables.scss';

.header {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  background-color: $gray-600;
  padding: 16px 16px;

  @include media-large {
    flex-direction: row;
    padding: 20px 96px;
  }

  @include media-medium {
    flex-direction: row;
    padding: 20px 48px;
  }
}

.title {
  font-size: 24px;

  @include media-medium {
    min-width: 300px;
  }
}
