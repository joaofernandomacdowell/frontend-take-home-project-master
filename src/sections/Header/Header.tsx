import React from 'react';

import styles from './Header.module.scss';
import { Container, Row } from '../../components/Grid';

const Header = (): JSX.Element => {
  return (
    <>
      <header className={styles.header}>
        <Container>
          <Row>
            <h1 className={styles.title}>Countries Rest API</h1>
          </Row>
        </Container>
      </header>
    </>
  );
};

export default Header;
