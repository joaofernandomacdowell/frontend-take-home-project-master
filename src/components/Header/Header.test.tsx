import React from 'react';
import { Provider } from 'react-redux';

import { render, screen } from '@testing-library/react';
import Header from './Header';

import store from '../../redux/store';

const header = render(
  <Provider store={store}>
    <Header />
  </Provider>
);

describe('Header', () => {
  describe('snapshot', () => {
    test('Should match render snapshot', () => {
      expect(Header).toMatchSnapshot();
    });
  });
  describe('component', () => {
    test('check h1 tag text', () => {
      const { getByText } = header;

      expect(
        screen.getByText('Countries Rest API')
      ).toBeInTheDocument();
    });
  });
});
