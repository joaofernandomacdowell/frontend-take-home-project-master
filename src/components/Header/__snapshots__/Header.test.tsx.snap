// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot Should match render snapshot 1`] = `[Function]`;
